nautobot/extras/models/groups.py

```diff
@@ -56,86 +57,235 @@ class DynamicGroup(OrganizationalModel):
 
     clone_fields = ["content_type", "filter"]
 
+    # This is used as a `startswith` check on field names, so these can be
+    # explicit fields or just substrings.
+    #
+    # Currently this means skipping "search", custom fields, and custom relationships.
+    #
+    # FIXME(jathan): As one example, `DeviceFilterSet.q` filter searches in `comments`. The issue
+    # really being that this field renders as a textarea and it's not cute in the UI. Might be able
+    # to dynamically change the widget if we decide we do want to support this field.
+    #
+    # Type: tuple
+    exclude_filter_fields = ("q", "cr", "cf", "comments")  # Must be a tuple
+
     class Meta:
         ordering = ["content_type", "name"]
 
     def __str__(self):
         return self.name
 
-    def get_queryset(self, **kwargs):
+    @property
+    def model(self):
+        """
+        Access to the underlying Model class for this group's `content_type`.
+
+        This class object is cached on the instance after the first time it is accessed.
+        """
+
+        if getattr(self, "_model", None) is None:
+            try:
+                model = self.content_type.model_class()
+            except models.ObjectDoesNotExist:
+                model = None
+
+            if model is not None:
+                self._set_object_classes(model)
+
+            self._model = model
+
+        return self._model
+
+    def _set_object_classes(self, model):
+        """
+        Given the `content_type` for this group, dynamically map object classes to this instance.
+        Protocol for return values:
+
+        - True: Model and object classes mapped.
+        - False: Model not yet mapped (likely because of no `content_type`)
+        """
+
+        # If object classes have already been mapped, return True.
+        if getattr(self, "_object_classes_mapped", False):
+            return True
+
+        # Try to set the object classes for this model.
+        try:
+            self.filterset_class = get_filterset_for_model(model)
+            self.filterform_class = get_form_for_model(model, form_prefix="Filter")
+            self.form_class = get_form_for_model(model)
+        # Fail closed.
+        except TypeError:
+            logger.debug("Failed to map object classes for model %s", model)
```

Comment:
Only a `debug` message? Is this something we expect during normal operation, or should this be a `warning`/`error` message instead?

Reply:
It's expected when `model `is None` such as when `self.content_type` is not yet set (new instances).